c++/src/H5DataSet.cpp

```diff
@@ -76,6 +76,21 @@ DataSet::DataSet(const DataSet &original) : H5Object(), AbstractDs(), id(origina
     incRefCount(); // increment number of references to this id
 }
 
+//--------------------------------------------------------------------------
+// Function:    DataSet assignment operator
+///\brief       Assignment operator: same HDF5 object as \a original
+///\param       original - IN: DataSet instance to copy
+// Programmer   Lee Newberg - 2021
```

Comment:
I'm not excited about programmer graffitti. I'm trying to remove author fields in comments when I find them as they are frequently meaningless. Developers should be using the history to determine this.

Reply:
I will be adding test for this and will take care of the issues you mentioned.  Thanks!